analisador-de-numeros.js

```diff
@@ -0,0 +1,117 @@
+let inputNumero = document.querySelector('#numero')
+let resultadoContainer = document.querySelector('#resultados')
+let mensagensContainer = document.querySelector('#mensagens')
+inputNumero.focus()
+
+let listaDeNumeros = []
+
+function inserirNovoNumero() {
+  let novoNumeroString = document.querySelector('#numero').value
+  let novoNumero = Number(novoNumeroString)
+  mensagensContainer.innerHTML = ''
+  resultadoContainer.innerHTML = ''
+  inputNumero.value = ''
+  inputNumero.focus()
+
+  try {
+    verificarSeONumeroEhValido(novoNumeroString)
+    adicionarNumeroNoArray(novoNumero)
+    mostrarNumeros(listaDeNumeros)
+  }
+  catch(erro) {
+    mensagensContainer.innerHTML = erro
+  }
+}
+
+function mostrarResultados() {
+  try {
+    if(listaDeNumeros.length === 0) throw 'Nenhum número foi adicionado ainda'
+    resultadoContainer.innerHTML = `Quantidade de números cadastrados: ${calcularAQuantidadeDeNumerosCadastrados(listaDeNumeros)} <br>`
+    resultadoContainer.innerHTML += `O menor número é: ${obterOMenorNumero(listaDeNumeros)} <br>`
+    resultadoContainer.innerHTML += `O maior número é: ${obterOMaiorNumero(listaDeNumeros)} <br>`
+    resultadoContainer.innerHTML += `A soma de todos os números é: ${calcularASomaDeTodosOsNumeros(listaDeNumeros)} <br>`
+    resultadoContainer.innerHTML += `A média de todos os números é: ${calcularAMediaDeTodosOsNumeros(listaDeNumeros)} <br>`
+  }
+  catch(erro) {
+    mensagensContainer.innerHTML = erro
+  }
+}
+
+function verificarSeONumeroEhValido(numeroString) {
+  let numero = Number(numeroString)
+  if(numeroString === '') {
+    throw 'O campo está vazio'
+  }
+  if(numero < 1) {
+    throw 'O número é menor que 1'
+  }
+  if(numero > 100) {
+    throw 'O número é maior que 100'
+  }
+  if(listaDeNumeros.indexOf(numero) != -1) {
+      throw 'Este número já foi adicionado'
+  }
+}
+
+function adicionarNumeroNoArray(numero) {
+  listaDeNumeros.push(numero)
+}
+
+function mostrarNumeros(listaDeNumeros) {
+  let numerosContainer = document.querySelector('#numeros')
+  numerosContainer.innerHTML = ''
+  for (posicao = 0; posicao < listaDeNumeros.length; posicao++) {
+    numerosContainer.innerHTML += `${listaDeNumeros[posicao]} | `
+  }
```

Comment:
Se você estiver usando JavaScript ES5, esse loop com posições pode ser substituído por:
```
listaDeNumeros.forEach(function (numero) {
  numerosContainer.innerHTML += `${numero} | `
});
```
Se você estiver usando ES6, dá pra usar a sintaxe `for of`:
```
for (numero of listaDeNumeros) {
  numerosContainer.innerHTML += `${numero} | `
}
```